src/main/java/org/broadinstitute/hellbender/tools/copynumber/utils/annotatedregion/SimpleAnnotatedIntervalWriter.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableWriter;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.util.List;
+
+
+/**
+ * Callers must call {@link #writeHeader} before {@link #add}.
+ *
+ * This class is not thread-safe.
+ */
+public class SimpleAnnotatedIntervalWriter implements AnnotatedIntervalWriter {
+
+    private SimpleTableWriter writer;
+    private FileWriter fileWriter;
+    private String contigColumnHeader;
+    private String startColumnHeader;
+    private String endColumnHeader;
+    private File outputFile;
+    private boolean hasHeaderBeenWritten = false;
+    private static final Logger logger = LogManager.getLogger(SimpleAnnotatedIntervalWriter.class);
+
+    private class SimpleTableWriter extends TableWriter<AnnotatedInterval> {
+
+        SimpleTableWriter(final Writer writer, TableColumnCollection tableColumns) throws IOException {
+            super(writer, tableColumns);
+        }
+
+        @Override
+        protected void composeLine(final AnnotatedInterval record, final DataLine dataLine) {
+            // First the Locatable info
+            dataLine.set(contigColumnHeader, record.getContig());
+            dataLine.set(startColumnHeader, record.getStart());
+            dataLine.set(endColumnHeader, record.getEnd());
+
+            // Now everything else.
+            record.getAnnotations().keySet().forEach(k -> dataLine.set(k, record.getAnnotationValue(k)));
+        }
+    }
+
+    /**
+     * Initialize this writer to the given output file.
+     *
+     * @param outputFile destination file.  Must be writeable.
+     */
+    public SimpleAnnotatedIntervalWriter(final File outputFile) {
+        Files.isWritable(outputFile.toPath());
+        this.outputFile = outputFile;
+    }
+
+    private void initializeForWriting(final String contigColumnName, final String startColumnName, final String endColumnName, final List<String> annotations) {
+        final List<String> finalColumnList = Lists.newArrayList(contigColumnName, startColumnName, endColumnName);
+        finalColumnList.addAll(annotations);
+        try {
+            fileWriter = new FileWriter(outputFile);
+
+            // By initializing writer to be based on fileWriter, writer.close will close the fileWriter as well.
+            writer = new SimpleTableWriter(fileWriter, new TableColumnCollection(finalColumnList));
+        } catch (final IOException ioe) {
+            throw new UserException.CouldNotCreateOutputFile(outputFile, "Could not create: " + outputFile.getAbsolutePath());
```

Comment:
Isn't this `new UserException.CouldNotCreateOutputFile(outputFile, ioe);`? It seems like the user would really want the exception in the output, as it could be a few different things.

Reply:
Made it a GATKException.  Now user will get runtime error details.

Done.